### PR TITLE
Add "Run with PowerShell" to ps1 Windows Explorer context menu

### LINKED ISSUE #94582: VSCode removes "Run with PowerShell" from Windows Explorer context menu
- VSCode Version:  latest
- OS Version:  Win10 v1909

We need an open issue for this issue (the super vscodebot is super aggressive in closing issues before they are fixed).  And this issue really needs to be fixed!

ref:
https://github.com/microsoft/vscode/issues/64815
https://github.com/PowerShell/vscode-powershell/issues/692
https://github.com/Microsoft/vscode/issues/6387

The cmds for 'edit in ISE' and 'run with PS' are missing missing on the PS context menu.

### PR DESCRIPTION
**Pull request by the-ress**
This adds "Run with PowerShell" to Windows Explorer context menu when .ps1 files are associated with VS Code.

This PR fixes #94582

### CODE REVIEW

**Comment by joaomoreno** [comment]
No I don't really think this is a good idea. That menu entry is there before VS Code gets installed. The fix for #94582 is to find out why it disappears.

**Comment by the-ress** [comment]
@joaomoreno I'm sorry, I should have included more details:

TLDR: It's fixed in the next Windows 10 version, but this PR might be a good solution for users that can't/won't update yet.

### Why is this happening

On a fresh Windows 10 installation, the default verbs for .ps1 files (Open, Run with PowerShell, and Edit) are a part of the default ProgID (`Microsoft.PowerShellScript.1`). VS Code creates its own separate ProgID during instalation (as it should) - `VSCode.ps1` - with only the Open verb.

image

image

Then, if user chooses to associate `.ps1` files with VS Code (using the Open With dialog), Windows associates the file extension with `VSCode.ps1` ProgID and the context menu won't show the verbs from `Microsoft.PowerShellScript.1`. This happens when associating `.ps1` files with any other application too.

### Windows 2004

While writing this, I noticed that in the next Windows 10 version (2004, build 19041) it's fixed on the system level by moving the Run with PowerShell, and Edit verbs to `HKEY_LOCAL_MACHINE\SOFTWARE\Classes\SystemFileAssociations\.ps1`. This makes these verbs show up in the menu regardless of which ProgID is associated with `.ps1` files.

image

### Fix for current/older Windows versions

While this issue will be fixed with the next Windows release, there are users that can't or won't update yet. It could be worth having a fix/workaround for them.

Possible solutions would be:
1. Add the Run with PowerShell verb to `VSCode.ps1`. This takes effect only when `.ps1` files are associated with VS Code. This is the approach I implemented in this PR. I could also add a condition so the fix/workaround is only applied on Windows 1909 or older, but even without it, the menu item won't be duplicated on 2004. (The `VSCode.ps1` verb overrides the one in `SystemFileAssociations`.)
2. Modify `HKEY_LOCAL_MACHINE\SOFTWARE\Classes\SystemFileAssociations\.ps1` during VS Code setup (or the same key in `HKEY_CURRENT_USER` for user-level installations). I don't think this is a good idea as it involves modifying shared system settings.

**Comment by joaomoreno** [comment]
@the-ress, that's a great analysis, thanks a lot!

Since Windows 10 is all about rolling forward, I don't think it's necessary to work around this in VS Code. It's fine to tell users to upgrade Windows 10, since the issue isn't really severe.

**Comment by the-ress** [comment]
I know firsthand there are developers who won’t be able to upgrade for a year or two due to corporate policies, but as you said, the issue isn’t really severe. I’ll post this fix/workaround as a `.reg` file in the issue in case someone is searching for it.
